addon/shield-setup.js

```diff
@@ -36,8 +58,60 @@ this.shieldSetup = (function () {
     }
   }, CHECK_SIDEBAR_PERIOD);
 
+  async function loadInstalledDate() {
+    let result = await browser.storage.local.get(["installedDate", "hasSeenMidwaySurvey"]);
+    if (result.installedDate) {
+      installedDate = result.installedDate;
+    } else {
+      installedDate = Date.now();
+      await browser.storage.local.set({installedDate});
+    }
+    if (result.hasSeenMidwaySurvey) {
+      hasSeenMidwaySurvey = result.hasSeenMidwaySurvey;
+    }
+  }
+
+  async function maybeOpenkMidwaySurvey() {
```

Comment:
Is maybeOpenkMidwaySurvey a typo or does the k indicate something meaningful? It's spelled the same way in the definition and where it's called, so it works.

Reply:
It's a misspelling, but autocomplete FTW! But fixed now...